core/src/test/java/org/elasticsearch/ElasticsearchExceptionTests.java

```diff
@@ -516,6 +518,41 @@ public void testFromXContentWithHeadersAndMetadata() throws IOException {
         assertThat(cause.getMetadata("es.index_uuid"), hasItem("_na_"));
     }
 
+    public void testThrowableToAndFromXContent() throws IOException {
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+
+        final Tuple<Throwable, ElasticsearchException> exceptions = randomExceptions();
```

Comment:
can we test what happens when one calls `setResources` passing in multiple ids? That's the one place where we need lists to work properly.